prow/config/config.go

```diff
@@ -310,6 +315,18 @@ type Branding struct {
 // PubSubSubscriptions maps GCP projects to a list of Topics.
 type PubsubSubscriptions map[string][]string
 
+// GitHubOptions allows users to control how prow application interact with
+// git, github, and github API's.
+type GitHubOptions struct {
+	// LinkURL allows users to override the default github link url for all plugins.
+	// If this option is not set, we use "https://github.com".
+	LinkURL string `json:"link_url,omitempty"`
+
+	// APILinkURL allows users to override the default github API link url for all plugins.
+	// If this option is not set, we use "https://api.github.com".
+	APILinkURL string `json:"api_link_url,omitempty"`
```

Comment:
The second point here hasn't been addressed yet: https://github.com/kubernetes/test-infra/pull/11273#discussion_r256467491
> * slices to allow for fallback as we do today

This is only needed for the APILinkURL which doesn't appear to be used yet. This value will need to be used by GitHub clients throughout Prow instead of relying on the `--github-endpoint` flags. I'd suggest removing this field for now and adding a TODO comment to add it later (and deprecate `--github-endpoint`) since that will be a substantial change that doesn't appear to be required by this PR.

Reply:
Removed APILinkURL variable and added a TODO for later PR.